src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_GENDER_CONSTRAINTS =
+            "Person gender should be a string of either 'Male', 'Female' or 'M', 'F'";
+    public static final String GENDER_VALIDATION_WORD1 = "Male";
+    public static final String GENDER_VALIDATION_WORD2 = "Female";
+    public static final String GENDER_VALIDATION_WORD3 = "M";
+    public static final String GENDER_VALIDATION_WORD4 = "F";
+
+    public final String value;
+
+    /**
+     * Validates given gender.
+     *
+     * @throws IllegalValueException if given gender string is invalid.
+     */
+    public Gender(String gender) throws IllegalValueException {
```

Comment:
perhaps here, we should catch "M/F" and "m/f" inputs and convert them to "Male/Female" to be assigned to value, instead of value possibly having 4 different string values

Reply:
agree!